lib/AST/NameLookup.cpp

```diff
@@ -1696,7 +1696,18 @@ bool DeclContext::lookupQualified(Type type,
     // error later.
     if (!decl->hasValidSignature())
       return true;
-    
+
+    if (auto *NTD = dyn_cast<NominalTypeDecl>(decl)) {
+      if (NTD->isGenericContext()) {
+        auto *DC = NTD->getDeclContext();
+        // Refuse to return results for situation when
+        // we have a nested type in generic context.
+        if (DC->getAsProtocolOrProtocolExtensionContext() ||
+            (DC->isLocalContext() && DC->isGenericContext()))
```

Comment:
openGeneric() should already support generic types inside local contet via bindArchetypesFromContext(), and eventually I want the rest of the compiler to support it too. Please don't make this change.